extensions/GitExtension/FileExplorerGitIntegration/Models/GitCommandRunner.cs

```diff
@@ -0,0 +1,229 @@
+
+    private string GitExeInstallPath { get; set; } = string.Empty;
+
+    private readonly FileSystemWatcher fileWatcher;
+
+    private readonly ILogger log = Log.ForContext<GitCommandRunner>();
+
+    public GitCommandRunner(string? path)
+    {
+        if (RuntimeHelper.IsMSIX)
+        {
+            GitExecutableConfigOptions = new GitExecutableConfigOptions
+            {
+                GitExecutableConfigFolderPath = ApplicationData.Current.LocalFolder.Path,
+            };
+        }
+        else
+        {
+            GitExecutableConfigOptions = new GitExecutableConfigOptions
+            {
+                GitExecutableConfigFolderPath = path ?? string.Empty,
+            };
+        }
+
+        fileService = new FileService();
+        ReadInstallPath();
+
+        fileWatcher = new FileSystemWatcher(GitExecutableConfigOptions.GitExecutableConfigFolderPath, GitExecutableConfigOptions.GitExecutableConfigFileName);
+        fileWatcher.NotifyFilter = NotifyFilters.LastWrite;
+        fileWatcher.Changed += OnFileChanged;
+        fileWatcher.EnableRaisingEvents = true;
+        log.Debug("FileSystemWatcher initialized for configuration file");
+    }
+
+    private void OnFileChanged(object sender, FileSystemEventArgs args)
+    {
+        if (args.Name == GitExecutableConfigOptions.GitExecutableConfigFileName)
+        {
+            ReadInstallPath();
+        }
+    }
+
+    public void ReadInstallPath()
+    {
+        GitExeInstallPath = fileService.Read<string>(GitExecutableConfigOptions.GitExecutableConfigFolderPath, GitExecutableConfigOptions.GitExecutableConfigFileName);
+    }
+
+    public bool IsGitExeInstallPathSet()
+    {
+        return !string.IsNullOrEmpty(GitExeInstallPath);
+    }
+
+    public GitCommandRunnerResultInfo ValidateGitExeInstallPath()
+    {
+        try
+        {
+            System.Diagnostics.Process.Start(GitExeInstallPath);
+            return new GitCommandRunnerResultInfo(ProviderOperationStatus.Success, null);
+        }
+        catch (Exception ex)
+        {
+            log.Error(ex, "Failed to start Git.exe at configured path");
+            return new GitCommandRunnerResultInfo(ProviderOperationStatus.Failure, "Failed to start Git.exe at configured path", string.Empty, ex, null);
+        }
+    }
+
+    public bool StoreGitExeInstallPath(string path)
+    {
+        log.Information("Setting Git Exe Install Path");
+        GitExeInstallPath = path;
+        fileService.Save(GitExecutableConfigOptions.GitExecutableConfigFolderPath, GitExecutableConfigOptions.GitExecutableConfigFileName, GitExeInstallPath);
+        log.Information("Git Exe Install Path stored successfully");
+        return true;
+    }
+
+    public bool DetectGit()
+    {
+        var gitExeFound = false;
+
+        if (!gitExeFound)
+        {
+            // Check if git.exe is present in PATH environment variable
+            try
+            {
+                System.Diagnostics.Process.Start("git.exe");
```

Comment:
I'm surprised this works.  I would think `UseShellExecute` needs to be true.  Does CreateProcess use EV's?

Reply:
Setting it to false indicates that the process is created directly from exe and does work in the current use case of invoking git with potential arguments. Can you please let me know what EVs stands for in this context. Thanks!

Reply:
EV = Environment Variable.

Reply:
Setting what to false?

Reply:
My understanding is that `Process.Start` uses the `PATH` env variable. With the caveat that the variable must be formatted in a way that .NET understands: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.start?view=net-8.0#system-diagnostics-process-start

`If you have a path variable declared in your system using quotes, you must fully qualify that path when starting any process found in that location. Otherwise, the system will not find the path. For example, if c:\mypath is not in your path, and you add it using quotation marks: path = %path%;"c:\mypath", you must fully qualify any process in c:\mypath when starting it.`